core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -1165,12 +1161,15 @@ private[history] case class LogInfo(
     logType: LogType.Value,
     appId: Option[String],
     attemptId: Option[String],
-    fileSize: Long)
+    fileSize: Long,
+    lastIndex: Option[Long],
```

Comment:
I missed to check this. Will check and fix if it causes problem via adding UT which loads LevelDB from Spark 2.4.4 and read more applications.

Reply:
I've crafted some test code to experiment this: 

```
    val newLogDir = new File("/spark-dist/releases/spark-2.4.4-bin-hadoop2.7/eventlogs")
    val localStoreDir = new File("/spark-dist/releases/spark-2.4.4-bin-hadoop2.7/history-dir")
    val newLocalStoreDir = new File(testDir, "localstore")
    newLocalStoreDir.mkdirs()
    FileUtils.copyDirectory(localStoreDir, newLocalStoreDir)

    val conf = createTestConf()
    conf.set(HISTORY_LOG_DIR, newLogDir.getAbsolutePath)
    conf.set(LOCAL_STORE_DIR, newLocalStoreDir.getAbsolutePath)

    val provider = new FsHistoryProvider(conf)
    provider.checkForLogs()
```

Here the directory directly points to the eventlogs/history local store directory in Spark 2.4.4 since metadata of listing contains the path and Spark ignores the content of local store if metadata doesn't match. If we want to include this to new UT, some modification might be needed. We tend to add UT if we want to ensure the new change is compatible with old version, so seem to be worth to add.

I've set breakpoint to `FsHistoryProvider.checkForLogs`, and observed lastIndex is set to `None` (so it seems to be safe), but `logType` is set to `null` which we don't expect.

Would we be better to increase `CURRENT_LISTING_VERSION` to deal with this, or manually set to `EventLogs` when it's null? Maybe better to deal with new PR though.

Reply:
Just realized that FsHistoryProvider does some care on `logType` being `null`. I wouldn't expect LogType.Value is null and feel safer to correct the value, but given now FsHistoryProvider accesses logType correctly, that's OK. I'd prefer to add UT to check the values of field of LogInfo when reading from Spark 2.4.4 to make sure we're not missing anything, but it's just me.

Reply:
You don't need to create a disk store with the old Spark version to test this. This is basically a test of deserializing an old JSON payload using `KVStoreScalaSerializer`; feed it JSON without the new field, and see what comes out of it.

Reply:
Good point. In other tests which we have modified something and worried about compatibility, we stored the old data and load from there so I thought about the same. Your suggestion would be pretty much simpler. Thanks!